src/utils/sourcemaps/inject.rs

```diff
@@ -0,0 +1,225 @@
+        fs::read_to_string(js_path).context(format!("Failed to open {}", js_path.display()))?;
+
+    if let Some(debug_id) = js::discover_debug_id(&file) {
+        debug!("File {} was previously processed", js_path.display());
+        report
+            .previously_injected
+            .push((js_path.to_path_buf(), debug_id));
+        return Ok(());
+    }
+
+    let Some(sourcemap_url) = js::discover_sourcemaps_location(&file) else {
+            debug!("File {} does not contain a sourcemap url", js_path.display());
+            report.skipped.push(js_path.to_path_buf());
+            return Ok(());
+        };
+
+    let sourcemap_path = js_path.with_file_name(sourcemap_url);
+
+    if !sourcemap_path.exists() {
+        warn!("Sourcemap file {} not found", sourcemap_path.display());
+        report.missing_sourcemaps.push(js_path.to_path_buf());
+        return Ok(());
+    }
+
+    let (debug_id, sourcemap_modified) = fixup_sourcemap(&sourcemap_path)
+        .context(format!("Failed to process {}", sourcemap_path.display()))?;
+
+    if sourcemap_modified {
+        report.sourcemaps.push((sourcemap_path, debug_id));
+    } else {
+        report.skipped_sourcemaps.push((sourcemap_path, debug_id));
+    }
+
+    fixup_js_file(js_path, debug_id).context(format!("Failed to process {}", js_path.display()))?;
+
+    report.injected.push((js_path.to_path_buf(), debug_id));
+
+    Ok(())
+}
+
+/// Appends the following text to a file:
+/// ```
+///
+/// <CODE_SNIPPET>[<debug_id>]
+/// //# sentryDebugId=<debug_id>
+///```
+/// where `<CODE_SNIPPET>[<debug_id>]`
+/// is `CODE_SNIPPET_TEMPLATE` with `debug_id` substituted for the `__SENTRY_DEBUG_ID__`
+/// placeholder.
+fn fixup_js_file(js_path: &Path, debug_id: DebugId) -> Result<()> {
+    let js_lines = {
+        let js_file = File::open(js_path)?;
+        let js_file = BufReader::new(js_file);
+        let js_lines: Result<Vec<_>, _> = js_file.lines().collect();
+        js_lines?
+    };
+
+    let mut sourcemap_comment = None;
+    let mut js_file = File::options().write(true).open(js_path)?;
+
+    for line in js_lines.into_iter() {
+        if line.starts_with("//# sourceMappingURL=") || line.starts_with("//@ sourceMappingURL=") {
+            sourcemap_comment = Some(line);
+            continue;
+        }
+        writeln!(js_file, "{line}")?;
+    }
+
+    let to_inject = CODE_SNIPPET_TEMPLATE.replace(DEBUGID_PLACEHOLDER, &debug_id.to_string());
+    writeln!(js_file)?;
+    writeln!(js_file, "{to_inject}")?;
+    writeln!(js_file, "{DEBUGID_COMMENT_PREFIX}={debug_id}")?;
+
+    if let Some(sourcemap_comment) = sourcemap_comment {
+        write!(js_file, "{sourcemap_comment}")?;
+    }
+
+    Ok(())
+}
+
+/// Fixes up a sourcemap file with a debug id.
+///
+/// If the file already contains a debug id under the `debugID` key, it is left unmodified.
+/// Otherwise, a fresh debug id is inserted under that key.
+///
+/// In either case, the value of the `debugID` key is returned.
```

Comment:
```suggestion
/// If the file already contains a debug id under the `debug_id` key, it is left unmodified.
/// Otherwise, a fresh debug id is inserted under that key.
///
/// In either case, the value of the `debug_id` key is returned.
```